tfjs-backend-webgpu/src/kernels/FromPixels.ts

```diff
@@ -64,37 +62,40 @@ export function fromPixels(args: {
       [pixels.width, pixels.height];
   const outShape = [height, width, numChannels];
 
-  if (env().getBool('WEBGPU_USE_IMPORT')) {
-    if (isVideo) {
-      return fromPixelsExternalImage({
-        externalImage: pixels as HTMLVideoElement,
-        backend,
-        attrs,
-        outShape,
-        useImport: true
-      });
+  const useImport = env().getBool('WEBGPU_USE_IMPORT') && isVideo;
+  const isVideoOrImage = isVideo || isImage;
+  const useKernel = isImageBitmap || isCanvas || isVideoOrImage;
+
+  let externalImage;
+  if (useImport) {
+    externalImage = pixels as HTMLVideoElement;
+  } else if (useKernel) {
+    if (isVideoOrImage) {
+      if (fromPixels2DContext == null) {
+        fromPixels2DContext = document.createElement('canvas').getContext('2d');
+      }
+      fromPixels2DContext.canvas.width = width;
+      fromPixels2DContext.canvas.height = height;
+      fromPixels2DContext.drawImage(
+          pixels as HTMLVideoElement | HTMLImageElement, 0, 0, width, height);
+      pixels = fromPixels2DContext.canvas;
     }
+    externalImage = pixels as HTMLCanvasElement | ImageBitmap;
   }
 
-  if (isVideo || isImage) {
-    if (fromPixels2DContext == null) {
-      fromPixels2DContext = document.createElement('canvas').getContext('2d');
-    }
-    fromPixels2DContext.canvas.width = width;
-    fromPixels2DContext.canvas.height = height;
-    fromPixels2DContext.drawImage(
-        pixels as HTMLVideoElement | HTMLImageElement, 0, 0, width, height);
-    pixels = fromPixels2DContext.canvas;
-  }
+  if (useImport || useKernel) {
```

Comment:
nit: ->`if (useKernel)`. I think `useImport` is true, `useKernel` must be also true.

Reply:
done